deeppavlov/models/kbqa/query_generator_online.py

```diff
@@ -0,0 +1,630 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import nltk
+from logging import getLogger
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.models.component import Component
+from deeppavlov.core.models.serializable import Serializable
+from typing import Tuple, List, Any
+from deeppavlov.models.kbqa.template_matcher import TemplateMatcher
+from deeppavlov.models.kbqa.entity_linking import EntityLinker
+from deeppavlov.models.kbqa.wiki_parser_online import WikiParserOnline, get_answer
+from deeppavlov.models.kbqa.rel_ranking_infer import RelRankerInfer
+
+log = getLogger(__name__)
+
+
+@register('query_generator_online')
+class QueryGeneratorOnline(Component, Serializable):
```

Comment:
Some class methods are similar to methods of `QueryGenerator`. To avoid code duplication ist's better to add parent class for `QueryGenerator` and `QueryGeneratorOnline`.

Reply:
I placed methods which are both in two query generators, in models/kbqa/utils.py